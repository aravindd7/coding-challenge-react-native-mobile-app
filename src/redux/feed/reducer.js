import {
  FEED_UPDATE__REQUEST,
  FEED_UPDATE__FAIL,
  FEED_UPDATE__SUCCESS,
  FEED_SECTION_CHANGE
} from "./actions";

const initialState = {
  section: "new", // new | top | hot | controversial
  state: "request", // request | fail | success
  error: undefined // Error message
};

const reducer = (state = initialState, action) => {
  switch (action.type) {
    case FEED_UPDATE__REQUEST:
      /* Reset state before fetch request */
      return { ...state, state: "request" };

    case FEED_UPDATE__FAIL:
      /* Update Fail fields */
      if (action.payload.error === undefined) return state;
      return { ...state, state: "fail", error: action.payload.error };

    case FEED_UPDATE__SUCCESS:
      /* Update Success fields */
      return { ...state, state: "success" };

    case FEED_SECTION_CHANGE:
      /* Chage Section */
      return { ...state, section: action.payload.section };

    default:
      return state;
  }
};

export default reducer;
