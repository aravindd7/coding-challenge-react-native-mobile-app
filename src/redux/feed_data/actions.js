export const FEED_DATA__UPDATE = "FEED_DATA__UPDATE";

export const feedData_Update = ({ data }) => ({
  type: FEED_DATA__UPDATE,
  payload: { data }
});
