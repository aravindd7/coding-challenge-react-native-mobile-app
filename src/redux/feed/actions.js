import { feedData_Update } from "../feed_data/actions";

export const FEED_UPDATE__REQUEST = "FEED_UPDATE__REQUEST";
export const FEED_UPDATE__FAIL = "FEED_UPDATE__FAIL";
export const FEED_UPDATE__SUCCESS = "FEED_UPDATE__SUCCESS";

export const FEED_SECTION_CHANGE = "FEED_SECTION_CHANGE";

export const changeSection = ({ section }) => {
  return (dispatch, getState) => {
    // TODO changeSection Action Creator
  };
};

export const feedUpdate = () => {
  return (dispatch, getState) => {
    // TODO feedUpdate Action Creator
  };
};

export const feedUpdate_Request = () => ({
  type: FEED_UPDATE__REQUEST,
  payload: {}
});

export const feedUpdate_Fail = ({ error }) => ({
  type: FEED_UPDATE__FAIL,
  payload: { error }
});

export const feedUpdate_Success = () => ({
  type: FEED_UPDATE__SUCCESS,
  payload: {}
});

export const feedSectionChange = ({ section }) => ({
  type: FEED_SECTION_CHANGE,
  payload: { section }
});
