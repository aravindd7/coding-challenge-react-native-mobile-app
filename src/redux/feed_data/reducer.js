import { FEED_DATA__UPDATE } from "./actions";

const initialState = []; // Array of news

const reducer = (state = initialState, action) => {
  switch (action.type) {
    case FEED_DATA__UPDATE:
      if (action.payload.data === undefined) return state;
      return action.payload.data;

    default:
      return state;
  }
};

export default reducer;
