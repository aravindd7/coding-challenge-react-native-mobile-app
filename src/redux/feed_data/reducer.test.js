import { FEED_DATA__UPDATE } from "./actions";

import reducer from "./reducer";

it("Reducer other action", async () => {
  const action = {
    type: "NO-ACTION"
  };
  const state = reducer(undefined, action);
  expect(state).toMatchSnapshot();
});

it("Reducer action: FEED_DATA__UPDATE", async () => {
  const action = {
    type: FEED_DATA__UPDATE,
    payload: {
      data: [
        { id: 1, url: "url-1", post: "post-1" },
        { id: 1, url: "url-1", post: "post-1" }
      ]
    }
  };
  const state = reducer(undefined, action);
  expect(state).toMatchSnapshot();
});

it("Reducer action: FEED_DATA__UPDATE with no data", async () => {
  const action = {
    type: FEED_DATA__UPDATE,
    payload: {}
  };
  const state = reducer(undefined, action);
  expect(state).toMatchSnapshot();
});
