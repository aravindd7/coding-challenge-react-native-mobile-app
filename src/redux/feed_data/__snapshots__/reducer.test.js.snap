// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Reducer action: FEED_DATA__UPDATE 1`] = `
Array [
  Object {
    "id": 1,
    "post": "post-1",
    "url": "url-1",
  },
  Object {
    "id": 1,
    "post": "post-1",
    "url": "url-1",
  },
]
`;

exports[`Reducer action: FEED_DATA__UPDATE with no data 1`] = `Array []`;

exports[`Reducer other action 1`] = `Array []`;
